src/fortran/dataset.F90

```diff
@@ -478,6 +480,33 @@ function get_tensor_type(self, name, ttype) result(code)
   code = get_tensor_type_c(self%dataset_ptr, c_name, name_length, ttype)
 end function get_tensor_type
 
+
+!> Retrieve the dimensions for a tensor into a supplied buffer, or receive the
+!! number of dimensions if the supplied buffer is too small
+function get_tensor_dims(self, name, dims, dims_length) result(code)
```

Comment:
For this one, I'm half wondering whether we want to be this strongly prescriptive about the `kind` of dims and dims_length. It would require an extra copy for both these variables, but they're so small that I can't think the hit would be significant.

So in the subroutine declaration it would just be:
```
integer, dimension(:), intent(inout) :: dims
integer,               intent(inout) :: dims_length

! Local variables
integer(kind=c_size_t), dimension(size(dims)), target :: c_dims
integer(kind=c_size_t) ::  c_dims_length
...
dims = c_dims
dims_length = c_dims_length

Reply:
Oh teh other thing about this, it follows the pattern of the rest of the Fortran SmartRedis client where we're trying NOT to make the user know about the `iso_c_binding` stuff in their own code.

Reply:
Sounds completely reasonable and more user-friendly